src/core/site.jl

```diff
@@ -40,11 +40,25 @@ function Site(;
     roof_squarefeet::Union{Real, Nothing} = nothing,
     min_resil_time_steps::Int=0,
     mg_tech_sizes_equal_grid_sizes::Bool = true,
+    CO2_emissions_reduction_min_pct::Union{Float64, Nothing} = nothing,
+    CO2_emissions_reduction_max_pct::Union{Float64, Nothing} = nothing,
+    bau_emissions_lb_CO2_per_year::Union{Float64, Nothing} = nothing,
+    bau_emissions_lb_NOx_per_year::Union{Float64, Nothing} = nothing,
+    bau_emissions_lb_SO2_per_year::Union{Float64, Nothing} = nothing,
+    bau_emissions_lb_PM25_per_year::Union{Float64, Nothing} = nothing,
+    bau_grid_emissions_lb_CO2_per_year::Union{Float64, Nothing} = nothing,
+    bau_grid_emissions_lb_NOx_per_year::Union{Float64, Nothing} = nothing,
+    bau_grid_emissions_lb_SO2_per_year::Union{Float64, Nothing} = nothing,
+    bau_grid_emissions_lb_PM25_per_year::Union{Float64, Nothing} = nothing,
+    renewable_electricity_min_pct::Real = 0.0,
+    renewable_electricity_max_pct::Union{Float64, Nothing} = nothing,
+    include_exported_elec_emissions_in_total::Bool = true,
+    include_exported_renewable_electricity_in_total::Bool = true,
     node::Int = 1, 
     )
 ```
 """
-struct Site
+mutable struct Site
```

Comment:
Why is Site now mutable? We should do what we can to keep it immutable.

Reply:
Resolving this comment because its due to https://github.com/NREL/REopt.jl/pull/28/files#r890278659